src/org/zaproxy/zap/view/DynamicFieldsPanel.java

```diff
@@ -25,19 +27,53 @@
 	private Map<String, ZapTextField> textFields;
 
 	public DynamicFieldsPanel(String[] requiredFields) {
-		this(requiredFields, new String[0]);
+		this(requiredFields, NO_FIELDS);
 	}
 
 	public DynamicFieldsPanel(String[] requiredFields, String[] optionalFields) {
-		super();
+		super(new GridBagLayout());
+
+		setFields(requiredFields, optionalFields);
+	}
+
+	/**
+	 * Sets the (required) fields that should be shown in the panel.
+	 * <p>
+	 * Any fields previously set are removed.
+	 *
+	 * @param requiredFields the required fields.
+	 * @throws IllegalArgumentException if the given argument is {@code null}.
```

Comment:
Should `see` the other method? 

At first I started to comment that the exception wasn't declared or thrown, but then I actually read the code :)